component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java

```diff
@@ -533,14 +540,28 @@ public Response getUserAvatarById(@Context UriInfo uriInfo,
 
         if (identity.isEnable() && !identity.isDeleted()) {
           InputStream stream = identityManager.getAvatarInputStream(identity);
+          byte[] avatarContent;
           if (stream != null) {
             /*
-             * As recommended in the the RFC1341
-             * (https://www.w3.org/Protocols/rfc1341/4_Content-Type.html), we
-             * set the avatar content-type to "image/png". So, its data would be
-             * recognized as "image" by the user-agent
-             */
+            * As recommended in the the RFC1341
+            * (https://www.w3.org/Protocols/rfc1341/4_Content-Type.html), we
+            * set the avatar content-type to "image/png". So, its data would be
+            * recognized as "image" by the user-agent
+            */
             builder = Response.ok(stream, "image/png");
+            int[] dimension = Utils.parseDimension(size);
+            try {
+              avatarContent = imageThumbnailService.getOrCreateThumbnail(identityManager.getAvatarFile(identity),
+                                                                         identity,
+                                                                         dimension[0],
+                                                                         dimension[1])
+                                                   .getAsByte();
+              if (avatarContent != null) {
+                builder = Response.ok(avatarContent, "image/png");
```

Comment:
Why making two calls of `Response.ok(stream, "image/png")` ? IMO, it should be managed by an `if...else` to ensure to either return a Thumbnail or the original image, without the return instruction to make sure to benifit from cache management (HTTP 304 when not modified). Besides, the eTag value must change depending on sent sizes to make sure that the HTTP 304 works for each size on its own.

```suggestion
            builder = Response.ok(stream, "image/png");
            int[] dimension = Utils.parseDimension(size);
            try {
              avatarContent = imageThumbnailService.getOrCreateThumbnail(identityManager.getAvatarFile(identity),
                                                                         identity,
                                                                         dimension[0],
                                                                         dimension[1])
                                                   .getAsByte();
              if (avatarContent != null) {
                builder = Response.ok(avatarContent, "image/png");
```

Reply:
Done

Reply:
We may avoid retrieving the avatar stream twice IMO, by testing on existence of `AvatarLastUpdated` field in both REST endpoints (`UserRestResourcesV1` & `SpaceRestResourcesV1`), by example:
```java
/* eTag Check - Start */
      builder = (eTag == null ? null : request.evaluatePreconditions(eTag));
      if (builder == null) {
/* eTag Check - Not matching Start */

// Retrieve one Stream content either from thumbnail, or original stream (when no size defined) or default avatar when the user/space doesn't have an avatar yet
if (lastUpdated == null) {// lastUpdated = identity.getAvatarLastUpdated() is already retrieved
  builder = getDefaultAvatarBuilder(); // Existing method From SpaceRestResourcesV1
} else {
  if (StringUtils.isNotBlank(size)) {
       int[] dimension = Utils.parseDimension(size);
      ...
     builder = Response.ok(avatarContent, "image/png");
  } else {
    InputStream stream = identityManager.getAvatarInputStream(identity);
     builder = Response.ok(stream, "image/png");
  }
}

/* eTag Check - Not matching End */

     }
/* eTag Check - End */

```

Besides, I think it would be better to not have a default size value in REST endpoint to be able to retrieve the original avatar image.

Reply:
We decide to put a default size at 45x45 because it is the major size required everywhere we use avatar : 
Stream, activity, whoonline, task ...
So, by using a default size, we ensure to treat all place the avatar is used.

But you right, there is no way to get the default avatar. 
So I propose to have a special size, "original", which return the original avatar if needed.
WDYT ?

About the other part "avoid retrieving the avatar stream twice IMO", I'm ok to change that when we agree on the original size

Reply:
Yep, totally agree, thanks !

Reply:
original avatar will be returned width `&size=0x0`

Reply:
Thanks for the precision @hakermi 
Is it sufficient @boubaker ?

Reply:
Yep, for the default value retrieved by `size=0x0` instead of using `original`, but we may need to make this as default value in REST endpoint instead of `45x45` (no need if it's too much modifications) & explain in doc that using 0x0 will retrieve the default value + a Unit Test to prevent it from regressing in future evolution.

Reply:
Done : add information on how to get original size, add unit test to be sure to prevent regression
Reorganize code to be more efficient when reading input stream : do not read default image if not needed, do not read original image if not needed.